src/search.cpp

```diff
@@ -137,7 +137,8 @@ Xapian::Query parse_query(Xapian::QueryParser* query_parser, std::string qs, int
 
     if (suggestion_mode && !query.empty()) {
       Xapian::Query subquery_phrase, subquery_anchored;
-      query_parser->set_default_op(Xapian::Query::op::OP_AND);
+      query_parser->set_default_op(Xapian::Query::op::OP_OR);
+      query_parser->set_stemming_strategy(Xapian::QueryParser::STEM_NONE);
```

Comment:
It's `set_stemming_strategy(Xapian::QueryParser::STEM_NONE)` that fixes the failure of the `Suggestion` unit-tests broken by the original "bad" fix. Is there any reason for using `OP_OR` instead of `OP_AND`? What negative effects may disabling of stemming have in this case?

Reply:
The stemmed terms do not have a `positionList` associated with them. They are just words, so they cannot be used for an `OP_PHRASE`. The bad fix used `OP_AND` which enforced stemming. Hence, the bad fix failed the older tests.

Disabling stemming won't have any effect at all compared to before this PR. It's by default disabled when we used `OP_PHRASE` as the default operator.

Using `OP_AND` instead of `OP_OR` won't have any significant impact. I felt that it is a better indicator of our motive here that we are only using it to get proper tokenizing into separate terms and the actual work is done by `OP_PHRASE` subquery.